ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/fullsync/FullSyncTargetManager.java

```diff
@@ -50,10 +53,28 @@
       final LabelledMetric<OperationTimer> ethTasksTimer) {
     super(config, protocolSchedule, protocolContext, ethContext, syncState, ethTasksTimer);
     this.config = config;
+    this.protocolContext = protocolContext;
     this.ethContext = ethContext;
     this.syncState = syncState;
   }
 
+  @Override
+  protected Optional<SyncTarget> finalizeSelectedSyncTarget(final SyncTarget syncTarget) {
+    final BlockHeader commonAncestor = syncTarget.commonAncestor();
+    if (protocolContext
+        .getWorldStateArchive()
+        .isWorldStateAvailable(commonAncestor.getStateRoot())) {
+      return Optional.of(syncTarget);
+    } else {
+      LOG.warn(
+          "Disconnecting {} because common ancestor at block {} is prior to our fast sync pivot block",
```

Comment:
For future-proofing, might be better to have a more generic message.  Something like "Disconnecting {} because world state is unavailable at common ancestor block {}"

Reply:
Yeah I went back and forth on how specific to be.  I suspect this area needs reworking as soon as we have a different way to wind up with blocks without state (e.g. pruning) anyway but probably better to be a bit generic.